src/app/rpmostree-compose-builtin-tree.c

```diff
@@ -836,7 +838,13 @@ impl_install_tree (RpmOstreeTreeComposeContext *self,
    * if one uses `check-passwd: { "type": "previous" }`. There are a few other optimizations
    * too, e.g. using the previous SELinux policy in unified core. Also, we might need the
    * commit *object* for next version incrementing. */
-  if (self->ref)
+  if (opt_previous_commit)
+    {
+      if (!ostree_repo_resolve_rev (self->repo, opt_previous_commit, FALSE,
+                                    &self->previous_checksum, error))
+        return FALSE;
+    }
+  else if (self->ref)
```

Comment:
Out of curiosity, from the original code, what is the semantic/meaning of this conditional?
Is there a legit case where neither branches are taken? If not, can we check for the `!(opt_previous_commit || self->self)` case upfront and turn this one into a plain `else`?

Reply:
> Is there a legit case where neither branches are taken?

RHCOS doesn't specify a ref (or at least soon won't), and if there isn't a previous build, then there's no previous commit to compare against. So then neither branches would be taken.